test/type/string_test.exs

```diff
@@ -0,0 +1,91 @@
+defmodule Litmus.Type.StringTest do
+  use ExUnit.Case
+
+  alias Litmus.Type
+
+  test "tests protocol implementation for String data type" do
+    type = %Type.String{
```

Comment:
You can make this one line. Also, can you put all the variable declarations together without empty newlines separating them? imo as it is, they look too spread out.